scripts/update_e2e_snapshots.py

```diff
@@ -107,15 +107,18 @@ def wait_for_workflow_completion(
         conclusion = workflow_run.get("conclusion")
 
         if status == "completed":
-            if conclusion != "success":
-                print(f"The latest workflow run failed with status: {conclusion}")
+            if conclusion == "failure":
+                return workflow_run
+            else:
+                print(
+                    f"The latest workflow run completed with status: {conclusion}. "
+                    "The snapshot update is only working on failed runs."
```

Comment:
The reason is that snapshots only need to be updated when there was a failure, right? It might make sense to add this as a comment